google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/DefaultRoutingPolicy.java

```diff
@@ -25,26 +25,26 @@
 import com.google.errorprone.annotations.concurrent.GuardedBy;
 import com.google.protobuf.ByteString;
 import java.math.BigInteger;
-import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
 
 public class DefaultRoutingPolicy implements RoutingPolicy {
-  private final int numPartitions;
+  private final long numPartitions;
   private final CloseableMonitor monitor = new CloseableMonitor();
 
   @GuardedBy("monitor.monitor")
-  private int nextWithoutKeyPartition;
+  private long nextWithoutKeyPartition;
 
-  public DefaultRoutingPolicy(int numPartitions) throws ApiException {
+  public DefaultRoutingPolicy(long numPartitions) throws ApiException {
     checkArgument(numPartitions > 0, "Must have a positive number of partitions.");
     this.numPartitions = numPartitions;
-    this.nextWithoutKeyPartition = new Random().nextInt(this.numPartitions);
```

Comment:
I'd prefer you didn't do this- it implies that there's some sort of concurrency going on where there isnt. Thoughts?

Reply:
The main reason I changed this is that Random doesn't implement the method nextLong(long bound), so we'd have to either use nextInt(int bound) or nextLong() both of which are fairly awkward

Reply:
Nah the change is fine.